### PR TITLE
Add E2E job which build and deploy designate operators

### PR DESCRIPTION
**Pull request by bshewale**

### CODE REVIEW

**Comment by jparoly** [comment]
@bshewale I have a change I would like to make that requires this PR get resolved and merged.  When do you think you will be able to get back to this PR?  Thank you! :-)

**Comment by bshewale** [comment]
> @bshewale I have a change I would like to make that requires this PR get resolved and merged. When do you think you will be able to get back to this PR? Thank you! :-)

Hi, I am trying to resolve the pre-commit failure here https://github.com/openstack-k8s-operators/designate-operator/pull/40 once it passes thn we can merge this PR.

**Comment by Sandeepyadav93** [comment]
/lgtm

Thanks for working on this.

**Comment by Sandeepyadav93** [comment]
@viroel @bshewale interesting, controller image build failed but job didn't fail.


https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/38395/rehearse-38395-pull-ci-openstack-k8s-operators-designate-operator-main-designate-operator-build-deploy/1659179461792239616/artifacts/designate-operator-build-deploy/openstack-k8s-operators-gather/artifacts/get_all.log
~~~
pod/designate-operator-1-build                                        0/1     Error       0               21m
~~~
https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/38395/rehearse-38395-pull-ci-openstack-k8s-operators-designate-operator-main-designate-operator-build-deploy/1659179461792239616/artifacts/designate-operator-build-deploy/openstack-k8s-operators-build/build-log.txt
~~~
Trying to pull gcr.io/distroless/static:nonroot...
Getting image source signatures
error: build error: error building at STEP "RUN if [ -f $CACHITO_ENV_FILE ] ; then source $CACHITO_ENV_FILE ; fi ; CGO_ENABLED=0  GO111MODULE=on go build ${GO_BUILD_EXTRA_ARGS} -a -o ${DEST_ROOT}/manager main.go": error while running runtime: exit status 2
~~~

**Comment by Sandeepyadav93** [comment]
Image build is failing in Github action also with same error:

https://github.com/openstack-k8s-operators/designate-operator/actions/runs/4916349223/jobs/8779983287

@bshewale please reach out to designate core to fix the image build.

File: ci-operator/config/openstack-k8s-operators/designate-operator/openstack-k8s-operators-designate-operator-main.yaml [hunk 1]
```diff
@@ -6,6 +15,22 @@ images:
...
+  from: src
+  inputs:
+    cli:
+      paths:
+      - destination_dir: .
+        source_path: /usr/bin/oc
+    opm-builder:
+      paths:
+      - destination_dir: .
+        source_path: /bin/opm
```
Comment (marios) [inline ci-operator/config/openstack-k8s-operators/designate-operator/openstack-k8s-operators-designate-operator-main.yaml:32]: sanity check... is that right? you have /usr/bin/opm on line 22

**Review by marios — COMMENTED** [review]
sanity check inline

File: ci-operator/config/openstack-k8s-operators/designate-operator/openstack-k8s-operators-designate-operator-main.yaml (hunk 1 above)
Comment (marios) [reply ci-operator/config/openstack-k8s-operators/designate-operator/openstack-k8s-operators-designate-operator-main.yaml:32]: thanks and sorry for the extra step. at least now we know it should be this way ;)

**Review by marios — COMMENTED** [review]
stared for a while didnt spot something 

looks like the job was green also https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/38395/rehearse-38395-pull-ci-openstack-k8s-operators-designate-operator-main-designate-operator-build-deploy/1660636837393207296

but you have the work in progress tag can you remove that

**Comment by bshewale** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-designate-operator-main-precommit-check

**Comment by bshewale** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-designate-operator-main-precommit-check

**Comment by bshewale** [comment]
This PR is Depends-On: https://github.com/openstack-k8s-operators/designate-operator/pull/52

**Comment by bshewale** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-designate-operator-main-precommit-check

**Comment by Sandeepyadav93** [comment]
@bshewale 

Hey o/

https://github.com/openstack-k8s-operators/openstack-operator/blob/main/config/samples/core_v1beta1_openstackcontrolplane.yaml

^^ we use this CR, I don't see designate in the example PR, Could you please check with designate cores about their plan to integrate designate in openstack-operator.

https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/38395/rehearse-38395-pull-ci-openstack-k8s-operators-designate-operator-main-designate-operator-build-deploy/1671515077078945792/artifacts/designate-operator-build-deploy/openstack-k8s-operators-gather/artifacts/pods/

I only see designate operator pods and no designate related service.

**Comment by jparoly** [comment]
needs a rebase? @bshewale

**Comment by viroel** [comment]
As @Sandeepyadav93 mentioned before, designate operator and service are not being deployed in this job. The job is only building the operator. It will need some work to add it on OpenStack CR[1]

[1] https://github.com/openstack-k8s-operators/openstack-operator/blob/main/config/samples/core_v1beta1_openstackcontrolplane.yaml

**Comment by viroel** [comment]
controller managers from latest run:
https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/38395/rehearse-38395-pull-ci-openstack-k8s-operators-designate-operator-main-designate-operator-build-deploy/1671515077078945792/artifacts/designate-operator-build-deploy/openstack-k8s-operators-gather/artifacts/controller-manager/

**Comment by jparoly** [comment]
Would it make sense to add the kuttl test execution in this PR as well or do so in another PR?

**Comment by viroel** [comment]
> Would it make sense to add the kuttl test execution in this PR as well or do so in another PR?

It would be better if you create another one. This one should not merge until designate is integrate with openstack-operator.
You could create a new job that only build and run kuttl tests, without deploy step. I think that would add value to the repo.

**Comment by bshewale** [comment]
As discussed in above comments we can not add the e2e designate operator build and deploy job until the designate is not integrate with openstack-operator so instead of that we can add the kuttl test and that i have added in PR: https://github.com/openshift/release/pull/43105 . We will close this PR and once the designate is integrate with openstack-operator we will re-open the this PR or will work on adding the e2 job.

**Pull request closed without merging**